File: pset1/mario/less/mario.c
#include <stdio.h>
#include <cs50.h>

int main(void)
{
    int d = -1;

    // Run until input is number 0 - 23
    while (d > 23 || d < 0)
    {
        d = get_int("Height: ");
    }

    // Main loop starting with 2 pieced step
    int steps = 2;
    for (int i = 0; i < d; i++)
    {
        int spaces = d - steps;
        // Spaces loop
        for (int m = 0; m <= spaces; m++)
        {
            printf(" ");
        }

        // Hashes loop
        for (int n = 0; n < steps; n++)
        {
            printf("#");
        }
        steps++;
        printf("\n");
    }
}

File: pset6/mario/less/mario.py
# Prints mario pyramid of selected height on screen

from cs50 import get_int

# Ask for positive integer up to 23
while True:
    height = get_int("Height: ")
    if height >= 0 and height < 24:
        break

# define steps
steps = 2

# main pyramid loop
for i in range(height):

    # calculate space count
    spaces = (height - steps) + 1

    # print spaces on screen
    for space in range(spaces):
        print(" ", end="")
    # print hashes on screen
    for step in range(steps):
        print("#", end="")

    # increment steps
    steps += 1
    print()